controlplane/rosa/api/v1beta2/rosacontrolplane_types.go

```diff
@@ -249,6 +260,24 @@ type RosaControlPlaneSpec struct { //nolint: maligned
 	// ClusterRegistryConfig represents registry config used with the cluster.
 	// +optional
 	ClusterRegistryConfig *RegistryConfig `json:"clusterRegistryConfig,omitempty"`
+
+	// AutoNode set the AutoNode mode and AutoNode role ARN.
+	// +optional
+	AutoNode *AutoNode `json:"autoNode,omitempty"`
+}
+
+// AutoNode set the AutoNode mode and AutoNode role ARN.
+type AutoNode struct {
+	// AutoNode mode specifies the mode for the AutoNode. Setting Enable/Disable mode will allows/disallow karpenter AutoNode scaling.
+	// +kubebuilder:validation:Enum=Enabled;Disabled
+	// +kubebuilder:default=Disabled
+	// +optional
+	Mode AutoNodeMode `json:"mode,omitempty"`
+
+	// AutoNode role ARN that has the IAM Policy and cluster-specific Role that gives permissions to the Karpenter controller.
+	// The role must be attached with the same OIDC-ID that is used with the ROSA-HCP cluster.
+	// +optional
+	RoleARN string `json:"roleARN,omitempty"`
```

Comment:
Since this will be exposed as a YAML field, the doc should start with `roleARN`, rather than the usual Go convention.